src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -360,9 +360,9 @@ public static bool IsUnnecessaryCast(this CastExpressionSyntax cast, SemanticMod
                 // Explicit reference conversions can cause an exception or data loss, hence can never be removed.
                 return false;
             }
-            else if (expressionToCastType.IsExplicit && expressionToCastType.IsNumeric && IsInExplicitCheckedOrUncheckedContext(cast))
+            else if (expressionToCastType.IsExplicit && expressionToCastType.IsNumeric)
```

Comment:
`IsInExplicitCheckedOrUncheckedContext` had exactly 1 reference, which means its now dead code. should be removed ?